x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/downsample/DownsampleTransportFailureTests.java

```diff
@@ -0,0 +1,414 @@
+        )
+    );
+
+    @Override
+    protected Collection<Class<? extends Plugin>> nodePlugins() {
+        return List.of(
+            LocalStateCompositeXPackPlugin.class,
+            Rollup.class,
+            AggregateMetricMapperPlugin.class,
+            DataStreamsPlugin.class,
+            IndexLifecycle.class
+        );
+    }
+
+    @Override
+    protected Collection<Class<? extends Plugin>> getMockPlugins() {
+        return List.of(MockTransportService.TestPlugin.class, TestSeedPlugin.class);
+    }
+
+    @Before
+    public void setup() {
+        startTime = LocalDateTime.parse("2020-09-09T18:00:00").atZone(ZoneId.of("UTC")).toInstant().toEpochMilli();
+        endTime = LocalDateTime.parse("2020-09-09T18:59:00").atZone(ZoneId.of("UTC")).toInstant().toEpochMilli();
+    }
+
+    @Override
+    public Settings indexSettings() {
+        return Settings.builder()
+            .put(super.indexSettings())
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 2)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+            .put(IndexSettings.MODE.getKey(), IndexMode.TIME_SERIES)
+            .putList(IndexMetadata.INDEX_ROUTING_PATH.getKey(), List.of("dim1"))
+            .put(
+                IndexSettings.TIME_SERIES_START_TIME.getKey(),
+                DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.formatMillis(Instant.ofEpochMilli(startTime).toEpochMilli())
+            )
+            .put(
+                IndexSettings.TIME_SERIES_END_TIME.getKey(),
+                DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER.formatMillis(Instant.ofEpochMilli(endTime).toEpochMilli())
+            )
+            .build();
+    }
+
+    public XContentBuilder indexMapping() throws IOException {
+        final XContentBuilder mapping = jsonBuilder().startObject().startObject("_doc").startObject("properties");
+        mapping.startObject("@timestamp").field("type", "date").endObject();
+        mapping.startObject("dim1").field("type", "keyword").field("time_series_dimension", true).endObject();
+        mapping.startObject("dim2").field("type", "long").field("time_series_dimension", true).endObject();
+        mapping.startObject("gauge").field("type", "long").field("time_series_metric", "gauge").endObject();
+        mapping.startObject("counter").field("type", "double").field("time_series_metric", "counter").endObject();
+        mapping.endObject().endObject().endObject();
+        return mapping;
+    }
+
+    public boolean indexDocuments(final String indexName, final List<String> documentsJson) {
+        final BulkRequestBuilder bulkRequestBuilder = client().prepareBulk();
+        documentsJson.forEach(document -> bulkRequestBuilder.add(new IndexRequest(indexName).source(document, XContentType.JSON)));
+        return bulkRequestBuilder.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE).get().hasFailures() == false;
```

Comment:
We usually assert the result of the operation inside the method, instead of returning a true/false result to the caller. It would only make sense to return the result of the operation to the caller method, if the caller method would handle each case differently.

In this case I would just assert the result here and make it the whole test fail if bulk has any failures